pkg/storage/extract.go

```diff
@@ -0,0 +1,43 @@
+package storage
```

Comment:
FYI - i moved this file as i originally introduced a third extraction type and method but then didn't end up needing it. i left the new file as i found it to be a little nicer when centralized. if someone disagrees i can happily undo this change